### PR TITLE
Added Secret of Evermore support

### PR DESCRIPTION
**Pull request by black-sliver**
* Patch rom with `python worlds/soe/Patch.py path/to/patch.apsoe`
* Use https://github.com/black-sliver/ap-soeclient/ to connect to AP

### CODE REVIEW

File: worlds/soe/__init__.py
Comment (Berserker66) [inline worlds/soe/__init__.py:96]: topology_present should be False, until meaningful Regions are a thing.

File: worlds/soe/__init__.py
Comment (Berserker66) [inline worlds/soe/__init__.py:98]: might want to bump that to 1 for the PR, though I don't mind doing that post-merge myself.

File: worlds/soe/__init__.py
Comment (Berserker66) [inline worlds/soe/__init__.py:173]: I don't get why this is needed? I hope it isn't, as for each SoE world iterate over each location is not very efficient for large multiworlds.

Comment (black-sliver) [reply worlds/soe/__init__.py:173]: `self.restrict_item_placement` is always False at the moment. i used that for testing since item_rules only work for progression. i can get rid of the code before merging if you want.

will also do the other two changes before the merge

**Comment by Berserker66** [comment]
Oh yeah, and a setup guide for the webhosts' guide section would be important.

File: worlds/soe/__init__.py
Comment (Berserker66) [inline worlds/soe/__init__.py:138]: list(map(... might be better.

**Comment by black-sliver** [comment]
The setup guide depends on two things:
* where do we host the client (I would be fine with evermizer.com since that would allow to fix bugs immediately)
* do we pull in SM / SNIClient first and I try to add apsoe patches to that and/or should i build an apbp patcher in wasm and host wherever the client ist hosted?

**Comment by Berserker66** [comment]
Planned pull in of SNIClient is next weekend, lordlou is using the next week to do some more fixing. You can host the client on your page, that's fine. I can offer a spot on AP as backup/mirror if wanted.

**Comment by black-sliver** [comment]
Okay, then i'll try to write an apbp wasm patcher, fix the remaining client issue, host everything on my site, put that into the guide. Once SNIClient is merged we'll see how much time is left before the release and if it makes sense to put SoE in there.

**Comment by Berserker66** [comment]
Sounds good. Should be able to merge this once the guide is done then.

File: WebHostLib/static/assets/gameInfo/en_Secret of Evermore.md
Comment (Berserker66) [inline WebHostLib/static/assets/gameInfo/en_Secret of Evermore.md:1]: Yes, Super Metroid....

**Review by Berserker66 — CHANGES_REQUESTED** [review]

**Comment by black-sliver** [comment]
* apbpatcher is here: https://www.evermizer.com/apbpatch/ and should also work for z3 and sm
* client is here: https://www.evermizer.com/apbpatch/
* `gameInfo/en_Secret of Evermore.md` should be fine now
* `tutorial/secret-of-evermore/multiworld_en.md` may still need proof reading (i've linked to that document for friday's internal beta test, so we'll see if people figure it out)

**Comment by Berserker66** [comment]
Thank you!

**Pull request merged**
